api_server.py

```
@@ -251,6 +243,83 @@ def check_rules(available_crew: List[Dict]) -> Dict:
+        now = get_now()
+        delta = max_end_time - now
+        days = max(1, delta.days)
+        return days
+    except Exception as e:
+        logger.error(f"Error calculating max days: {e}")
+        return 3
+
+
+def run_scraper_task(max_days: int):
+    """Run the scraper in a background thread."""
+    global fetch_state
```

Ruff (0.15.9)

[warning] Using the global statement to update `fetch_state` is discouraged

(PLW0603)